lib/ansible/modules/cloud/amazon/aws_ses_identity_policy.py

```diff
@@ -0,0 +1,199 @@
+        required: true
+    policy:
+        description: A properly formated JSON sending authorization policy. Required when I(state=present).
+    state:
+        description: Whether to create(or update) or delete the authorization policy on the identity.
+        default: present
+        choices: [ 'present', 'absent' ]
+requirements: [ 'botocore', 'boto3' ]
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- name: add sending authorization policy to domain identity
+  aws_ses_identity_policy:
+    identity: example.com
+    policy_name: ExamplePolicy
+    policy: "{{ lookup('template', 'policy.json.j2') }}"
+    state: present
+
+- name: add sending authorization policy to email identity
+  aws_ses_identity_policy:
+    identity: example@example.com
+    policy_name: ExamplePolicy
+    policy: "{{ lookup('template', 'policy.json.j2') }}"
+    state: present
+
+- name: add sending authorization policy to identity using ARN
+  aws_ses_identity_policy:
+    identity: "arn:aws:ses:us-east-1:12345678:identity/example.com"
+    policy_name: ExamplePolicy
+    policy: "{{ lookup('template', 'policy.json.j2') }}"
+    state: present
+
+- name: remove sending authorization policy
+  aws_ses_identity_policy:
+    identity: example.com
+    policy_name: ExamplePolicy
+    state: absent
+'''
+
+RETURN = '''
+policies:
+    description: A list of all policies present on the identity after the operation.
+    returned: success
+    type: list
+    sample: [ExamplePolicy]
+'''
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import sort_json_policy_dict
+
+import json
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+    from botocore.config import Config
+except ImportError:
+    pass  # caught by imported HAS_BOTO3
+
+
+def get_identity_policy(connection, module, identity, policy_name):
+    try:
+        response = connection.get_identity_policies(Identity=identity, PolicyNames=[policy_name])
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg='Failed to retrieve identity policy {policy}'.format(policy=policy_name))
+    policies = response['Policies']
+    if policy_name in policies:
+        return policies[policy_name]
+    return None
+
+
+def create_or_update_identity_policy(connection, module):
+    identity = module.params.get('identity')
+    policy_name = module.params.get('policy_name')
+    required_policy = module.params.get('policy')
+    required_policy_dict = sort_json_policy_dict(json.loads(required_policy))
+
+    changed = False
+    policy = get_identity_policy(connection, module, identity, policy_name)
+    policy_dict = sort_json_policy_dict(json.loads(policy)) if policy else None
+    if required_policy_dict != policy_dict:
```

Comment:
[compare_policies()](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/ec2.py#L681) is a better way of checking if there are changes (it takes two dictionaries, makes them hashable for comparison and returns True if they are different). s3_bucket and iam_managed_policy were both using sort_json_policy_dict which introduced bugs.